packages-ts/gauntlet-terra-contracts/src/commands/contracts/ocr2/withdrawFunds.ts

```diff
@@ -0,0 +1,76 @@
+import { BN, logger, prompt } from '@chainlink/gauntlet-core/dist/utils'
+import {
+  AbstractInstruction,
+  BeforeExecute,
+  ExecutionContext,
+  instructionToCommand,
+} from '../../abstract/executionWrapper'
+import { CATEGORIES } from '../../../lib/constants'
+import { AccAddress } from '@terra-money/terra.js'
+
+type CommandInput = {
+  amount?: string
+  recipient?: string
+  all?: boolean
+}
+
+type ContractInput = {
+  amount: string
+  recipient: string
+}
+
+const makeCommandInput = async (flags: any, args: string[]): Promise<CommandInput> => {
+  if (flags.input) return flags.input as CommandInput
+  return {
+    amount: flags.amount,
+    recipient: flags.recipient,
+    all: flags.all,
+  }
+}
+
+const makeContractInput = async (input: CommandInput, context: ExecutionContext): Promise<ContractInput> => {
+  const amount = input.all
+    ? ((await context.provider.wasm.contractQuery(context.contract, 'link_available_for_payment' as any)) as any).amount
+    : input.amount
+  const recipient = input.recipient || context.wallet.key.accAddress
+  return {
+    amount: new BN(amount).toString(),
+    recipient,
+  }
+}
+
+const validateInput = (input: CommandInput): boolean => {
```

Comment:
Why still use this and not split it into separate validation? Even if just one validation fn (no splitting), I'd still rather we be consistent and use a single API (new one).